src/Fable.Transforms/FableTransforms.fs

```diff
@@ -260,20 +260,21 @@ module private Transforms =
         match e with
         | Let(ident, value, letBody) when (not ident.IsMutable) && isErasingCandidate ident ->
             let canEraseBinding =
-                match value, com.Options.Language with
```

Comment:
@alfonsogarciacaro You can leave the language check for Rust for now, just to make it pass, I'll fix it later.

Reply:
Thanks @ncave! At the end I've taken a different approach (see edited comment) bit unfortunately it's still breaking Rust :/

Reply:
@alfonsogarciacaro I'll take a look, but can't we just keep the bindingBetaReduction as it was for Rust only? (just temporary)?

Reply:
You're right, that'd probably be the best solution. Unfortunately there have been several changes in the FSharp2Fable and FableTransforms steps. I've tried to undo some of them but Rust was still failing so I couldn't pinpoint the problem.

I've [disabled the 4 failing Rust tests](https://github.com/fable-compiler/Fable/pull/3360/commits/24df43cacb457d963e67998c8683e084dced097a). Do you think that'd be ok for now to merge this PR until we can investigate? It looks like an issue with the closures created by `sprintf` and friends. I had a similar problem in Typescript, but it was fixed by [adjusting the ident type after propagating uncurrying](https://github.com/fable-compiler/Fable/blob/24df43cacb457d963e67998c8683e084dced097a/src/Fable.Transforms/FableTransforms.fs#L423).

Reply:
@alfonsogarciacaro Sure, not a problem, please merge and I'll take a look.

Reply:
Thanks for confirming @ncave! Given that it's only 4 related tests, hopefully it's easy to spot the problem. BTW, I just realized `kprintf` is failing sometimes for TypeScript too. I'll try to fix that, maybe it's related to the Rust regression.

There is still work to do but good news is we already have >1000 tests passing for TypeScript :tada:

Reply:
@alfonsogarciacaro That sounds great!

The real challenge for all the typed languages (TS, Dart, Rust) are the Apllicative tests.
Hopefully your lambda changes will make things a bit easier there.